editoast/src/models/timetable.rs

```diff
@@ -0,0 +1,65 @@
+use super::{Retrieve, TrainSchedule, TrainScheduleDetails};
+use crate::error::{InternalError, Result};
+use crate::tables::osrd_infra_timetable;
+use diesel::pg::PgConnection;
+use diesel::prelude::*;
+use diesel::result::Error as DieselError;
+use editoast_derive::EditoastError;
+use editoast_derive::Model;
+use serde::Serialize;
+use thiserror::Error;
+
+#[derive(Debug, PartialEq, Queryable, Identifiable, Serialize, Selectable, Model)]
+#[model(table = "osrd_infra_timetable")]
+#[model(retrieve)]
+#[diesel(table_name = osrd_infra_timetable)]
+pub struct Timetable {
+    #[diesel(deserialize_as = i64)]
+    pub id: i64,
+    #[diesel(deserialize_as = String)]
+    pub name: String,
+}
+
+#[derive(Debug, Error, EditoastError)]
+#[editoast_error(base_id = "timetable")]
+enum TimetableError {
+    #[error("Timetable '{timetable_id}', could not be found")]
+    #[editoast_error(status = 404)]
+    NotFound { timetable_id: i64 },
+}
+
+#[derive(Debug, PartialEq, Serialize)]
+pub struct TimetableWithSchedules {
+    #[serde(flatten)]
+    pub timetable: Timetable,
+    pub train_schedules: Vec<TrainScheduleDetails>,
+}
+
+impl Timetable {
+    /// Retrieves timetable with a specific id and its associated train schedules
+    pub fn with_train_schedules(
+        conn: &mut PgConnection,
+        timetable_id: i64,
+    ) -> Result<TimetableWithSchedules> {
```

Comment:
We can keep returning an option. The benefit of it is to move `TimetableError` to the view.